pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
-    def accelerator_types(self) -> List[str]:
-        return ["auto"] + list(_AcceleratorType)
-
-    @property
-    def precision_plugin(self) -> PrecisionPlugin:
-        if self._precision_plugin is None:
-            self._precision_plugin = self.select_precision_plugin()
-        return self._precision_plugin
-
-    def final_strategy(self) -> Strategy:
-        if self._strategy is None:
-            self._strategy = self.select_strategy()
-        self._strategy = self.resolve_strategy(self._strategy)
-        # attach checkpoint plugin to the training type plugin
-        if self._checkpoint_io is not None:
-            self._strategy.checkpoint_io = self._checkpoint_io
-        if (
-            isinstance(self._strategy_flag, Strategy) and self._strategy_flag._precision_plugin is None
-        ) or not isinstance(self._strategy_flag, Strategy):
-            precision_plugin = self.precision_plugin
-            if precision_plugin is not None:
-                self._strategy.precision_plugin = precision_plugin
-        if (isinstance(self._strategy_flag, Strategy) and self._strategy_flag.accelerator is None) or not isinstance(
-            self._strategy_flag, Strategy
-        ):
-            self._strategy.accelerator = self.select_accelerator()
-        return self._strategy
-
-    @property
-    def cluster_environment(self) -> ClusterEnvironment:
-        if self._cluster_environment is None:
-            self._cluster_environment = self.select_cluster_environment()
-        return self._cluster_environment
-
-    @property
-    def has_cpu(self) -> bool:
-        return True
-
-    @property
-    def use_cpu(self) -> bool:
-        return self._accelerator_type == _AcceleratorType.CPU
+                else:
+                    self._cluster_environment_flag = getattr(self._strategy_flag, "cluster_environment")
 
-    @property
-    def has_gpu(self) -> bool:
-        # Here, we are not checking for GPU availability, but instead if User has passed
-        # `gpus` to Trainer for training.
-        gpus = self.parallel_device_ids
-        if gpus is not None and len(gpus) > 0:
-            return True
-        return self._map_devices_to_accelerator(_AcceleratorType.GPU)
+            # TODO: RFC existing accel_conn doesn't handle this, should we add conflict check?
+            #   eg: parallel_device is torch.device(cpu) but accelerator=gpu
+            if hasattr(self._strategy_flag, "parallel_devices"):
+                if self._strategy_flag.parallel_devices:
+                    if self._strategy_flag.parallel_devices[0].type == "cpu":
+                        self._accelerator_flag = "cpu"
+                    if self._strategy_flag.parallel_devices[0].type == "cuda":
+                        self._accelerator_flag = "gpu"
 
-    @property
-    def use_gpu(self) -> bool:
-        return self._accelerator_type == _AcceleratorType.GPU and self.has_gpu
+        amp_type = amp_type.lower() if isinstance(amp_type, str) else None
```

Comment:
Can we move this logic within `AMPType.from_str` ?

Reply:
updated, you are correct, the amp_type.lower() wasn't required, it's already handled in AMPType.from_str